### PR TITLE
add github action for go-vuln-check

### PR DESCRIPTION
**Pull request by rabajaj0509**
This PR solves https://github.com/apache/skywalking/discussions/10483#discussioncomment-5460399. Now onwards all PRs in golang repositories will be scanned for vulnerabilities.

### CODE REVIEW

File: .github/workflows/go-vuln.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+name: go-vuln
+
+on: [push, pull_request]
+permissions: read-all
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-go@v2
```
Comment (kezhenxu94) [inline .github/workflows/go-vuln.yml:28]: Let's use v4 to eliminate warnings and leverage builtin caching

```suggestion
      - uses: actions/setup-go@v4
```

**Comment by Superskyyy** [comment]
Shouldn't this be enabled through a .lift/config.toml? @kezhenxu94 @rabajaj0509 I'm confused.

**Comment by kezhenxu94** [comment]
> Shouldn't this be enabled through a .lift/config.toml? @kezhenxu94 @rabajaj0509 I'm confused.

I'm confused too, I have asked here https://github.com/apache/skywalking/discussions/10483#discussioncomment-5460399 about what's their outcome

**Comment by Superskyyy** [comment]
Let's hold this pr for a bit.

**Comment by Superskyyy** [comment]
Closing as sonatype has adopted the check as native pr bot. Following up action will be setting up sonatype in all go repos.

**Pull request closed without merging**